### ISSUE TITLE
[FP] `InsufficientTypesControlInspection`

### ISSUE THREAD

**Issue by jdreesen** [kalessil/phpinspectionsea#2006]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate), 2026.1 |
| Language level | PHP 8.1                         |

Current behaviour
--

`$this->max` is `int|float|null`
`$min` is `int|float`

For code like:

```php
if (isset($this->max) && $this->max < $min)
```

or:

```php
if (null !== $this->max && $this->max < $min)
```

EA warns:
> [EA] In multiple cases the result can be evaluated as false, consider hardening the condition.

<img width="913" height="78" alt="Image" src="https://github.com/user-attachments/assets/24e5ef60-f64b-4791-abfb-c03928d1fcc4" />

<img width="982" height="78" alt="Image" src="https://github.com/user-attachments/assets/9dcac2b7-fba7-401b-88ce-ed63ba6ff9fc" />

Expected behaviour
--

Inspection does not trigger a warning. Or do I have overlooked something?

Environment details
---

PhpStorm 2025.3.2
Build #PS-253.30387.85, built on January 21, 2026
Source revision: 790528c506585

Runtime version: 21.0.9+10-b1163.94 amd64 (JCEF 137.0.17)
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Toolkit: sun.awt.X11.XToolkit
Linux 6.8.0-100-generic
Ubuntu 24.04.4 LTS; glibc: 2.39
GC: G1 Young Generation, G1 Concurrent GC, G1 Old Generation
Memory: 10240M
Cores: 16
Current Desktop: ubuntu:GNOME

**Comment by ea-inspections-team** [comment]
Thank you for reporting @jdreesen!
The false positive has been resolved.
